src/classes/classes.js

```diff
@@ -0,0 +1,242 @@
+const maxItemsToDisplay = 5;
+const CACHE = {};
+
+function updateCache(id, searchTerm, data) {
+  CACHE[id] = CACHE[id] || {};
+  CACHE[id][searchTerm] = data;
+}
+
+function itemFromCache(id, searchTerm) {
+  if (CACHE[id] && CACHE[id][searchTerm]) {
+    return CACHE[id][searchTerm];
+  }
+
+  return null
+}
+
+// add event listener to the search field input
+document.getElementById("searchField").addEventListener("input", function() {
+  onSearchFieldInputChange();
+});
+
+// data feed setup
+let dataFeeds = {
+  reddit: {
+    id: "reddit",
+    url:
+      "https://api.reddit.com/api/subreddit_autocomplete_v2.json?limit=10&include_over_18=false&query=",
+    normaliseData: function(data) {
+      let normalisedData = [];
+      var data = data.data;
```

Comment:
```suggestion
    normaliseData: function(data) {
      let normalisedData = [];
      data = data.data;
```

☝️ I think this should be OK, and avoids the ugly `var`.